examples/vl3_basic/vl3-nse/cmd/service_registry.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/tiswanso/examples/api/serviceregistry"
+	"google.golang.org/grpc"
+	"strconv"
+	"strings"
+)
+
+const (
+	POD_NAME = "podName"
+	SERVICE_NAME = "service"
+	PORT = "port"
+	CLUSTER_NAME = "clusterName"
+)
+
+type validationErrors []error
+
+type ServiceRegistry interface {
+	RegisterWorkload(clusterName, podName, name, seviceName, connDom string, ipAddr []string, ports []int32) error
```

Comment:
add context here and use the right context when requesting the service instead of using `context.Backgroud()`

Reply:
done